packages/delegation-process/src/services/delegationProducerService.ts

```diff
@@ -95,6 +100,85 @@ export function delegationProducerServiceBuilder(
 
       return delegation;
     },
+    async approveProducerDelegation(
+      delegateId: TenantId,
+      delegationId: DelegationId,
+      correlationId: string
+    ): Promise<void> {
+      const delegationWithMeta = await readModelService.getDelegationById(
+        delegationId
+      );
+
+      const { data: delegation, metadata } = assertDelegationExists(
+        delegationId,
+        delegationWithMeta
+      );
```

Comment:
In this [PR](https://github.com/pagopa/interop-be-monorepo/pull/1100/files#diff-3fdedaa0c622c796919943910795a21c52deb912222479f41b453e08014687f2R49) I handled the same assertion requirement using a fetch function, like other approach used in existents services. Evaluate if a function like this could be usefull to future usages, otherwise leave as is.